rtl/eclrtl/rtldynfield.cpp

```diff
@@ -417,6 +555,20 @@ class CRtlFieldTypeDeserializer : public CInterfaceOf<IRtlFieldTypeDeserializer>
         types.setValue(name, type);
         return type;
     }
+    const RtlTypeInfo *lookupType(unsigned idx)
+    {
+        VStringBuffer key("%d", idx);
+        const RtlTypeInfo ** found = types.getValue(key);
+        if (found)
+            return *found;
+        throwUnexpected();
+    }
+    void addType(const RtlTypeInfo *type, unsigned idx)
+    {
+        VStringBuffer key("%d", idx);
```

Comment:
Alternatively could  have an array of types - may be worth commenting if want to revisit in the future.